Add clarification to differentiate between the blog URL and the RSS(feed URL) during sign-up

cychu42 opened the pull request:
<!--
Thanks for sending a pull request!
- if this is your first time, please ensure you have read through our contributor guide: https://github.com/Seneca-CDOT/telescope/blob/master/docs/CONTRIBUTING.md
-->

## Issue This PR Addresses
USAGE: #3646 
This PR addresses one of the issues raised in the issue.
<!--
1. Automatically close the issue when this PR is merged
    USAGE: Fixes #<issue number>
2. If your PR addresses an issue but does not close it
    USAGE: #<issue number> <reason>(i.e. This issue was worked on by @user and myself and his PR should close the issue.)
-->

## Type of Change

<!-- bug fix, feature, documentation, UI, etc. -->

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Description

![image](https://user-images.githubusercontent.com/83015577/202260230-e7d3d75d-1e2b-4c51-9b04-952954af9d1e.png)
The issue was that some users experience confusion between a blog URL (blog's root HTML) and select the RSS (blog's feed XML) during sign-up. This PR adds 2 sentences to the prompt to clarify the difference, in `src\web\app\src\components\SignUp\Forms\BlogFeeds.tsx`. 

## Steps to test the PR

Test this by going through the sign-up process to view the Blog Feeds page.
During testing, you can use test account `user1` with password `user1pass`.
You can enter any Github account name to get through the page before this page.

## Checklist

<!-- Before submitting a PR, address each item -->

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

NeilAn99 commented:
I think it might be nice to provide an example of a blog URL, so users who don't know what HTML and XML are know what to provide.

cychu42 commented:
Maybe, but there are many different sites, so I'm not sure it's practical to list them all or useful to list only one.

NeilAn99 commented:
I think maybe one example would be fine, I think most people use dev.to. But just a simple example, something like:

Blog URL: https://dev.to/user
vs
Blog feed URL: https://dev.to/feed/user

TueeNguyen commented:
Agree with @NeilAn99, let's add examples of blog url and feed url in 2 separate lines

cychu42 commented:
OK, how about this?
![image](https://user-images.githubusercontent.com/83015577/202331834-d57aa358-93d4-4a0e-9652-9ef99a0c54bb.png)

batunpc commented on src/web/app/src/components/SignUp/Forms/RSSFeeds.tsx:
```diff
@@ -143,6 +143,8 @@ type RSSFeedsFormProps = {
   };
   prompt: string;
+  promptExample1: string;
+  promptExample2: string;
```
Instead of `promptExample1` and `promptExample2`, won't it be better to take an array with prop example?

batunpc commented on src/web/app/src/components/SignUp/Forms/ChannelFeeds.tsx:
```diff
@@ -8,6 +8,8 @@ const ChannelFeeds = () => {
       title="YouTube and Twitch"
       prompt="OPTIONAL: Enter your YouTube and/or Twitch channel"
+      promptExample1=""
+      promptExample2=""
```
Here making the prop optional and assigning a default value like empty list , ensure that nothing will break.

batunpc commented on src/web/app/src/components/SignUp/Forms/RSSFeeds.tsx:
```diff
@@ -245,7 +249,13 @@ const RSSFeeds = connect<RSSFeedsFormProps, SignUpForm>(
+          <h2 className={classes.helpText}>
+            {prompt}
+            <br />
+            {promptExample1}
```
Having an array will make these dynamic and reusable

batunpc reviewed (changes requested):
@cychu42 Your implementation looks good!
I have left some initial review that we could address

cychu42 commented:
@batunpc Good ideas! I made some changes accordingly.
The only thing I didn't do is assigning a default value, because it doesn't look like the existing code does it for other variables of type `RSSFeedsFormProps` , but I did make the `promptExamples` array optional and make sure nothing would render if there's no value assigned. 
For instance, the next page renders using the same type without any example.
![image](https://user-images.githubusercontent.com/83015577/202739449-afd518af-c215-426d-9858-543d0caf51f1.png)

batunpc reviewed (approved):
Thank you @cychu42 , great job 👍

cychu42 commented:
Thank you all for the reviews. I just rebased and squashed the commits.
I guess I'm now waiting for a reviewer with write access to allow merge to happen.

manekenpix commented on src/web/app/src/components/SignUp/Forms/RSSFeeds.tsx:
```diff
@@ -245,7 +247,16 @@ const RSSFeeds = connect<RSSFeedsFormProps, SignUpForm>(
+          <h2 className={classes.helpText}>
+            {prompt}
+            {promptExamples?.map((example) => (
+              <>
+                <br />
+                {example}
+              </>
```
```suggestion
              <div>{example}</div>
```

manekenpix commented:
Last minute suggestion:

The text looks a bit bloated and hard to read, what about something more simple? maybe a basic but clear indication of what valid and non valid blog URLs are:

![image](https://user-images.githubusercontent.com/23108901/202873026-ac1a1b3b-818d-48bf-9058-6cfe89b59d66.png)

It doesn't have to be exactly this ^, but maybe something a bit cleaner would be better for users? It could also include links to blog URL vs feed URL explanations or definitions, or Blog vs Post (some people think a post is a blog).
I'm also ok if the team decides that the changes proposed in this PR are the way to go.

cychu42 commented:
What if we use some shorter text instead:
> Enter your blog root URL (HTML) and select the blog's RSS feed URL (XML) you want to use in Telescope, such as:
> Blog root URL: https://dev.to/user
> RSS feed URL: https://dev.to/feed/user

The 2 examples can be smaller texts like you suggested.

manekenpix commented:
> What if we use some shorter text instead:
> 
> > Enter your blog root URL (HTML) and select the blog's RSS feed URL (XML) you want to use in Telescope, such as:
> > Blog root URL: https://dev.to/user
> > RSS feed URL: https://dev.to/feed/user
> 
> The 2 examples can be smaller texts like you suggested.

Yeah, I think that'd work.

Myrfion commented on src/web/app/src/components/SignUp/Forms/BlogFeeds.tsx:
```diff
@@ -7,7 +7,9 @@ const BlogFeeds = () => {
       title="Blog and RSS"
-      prompt="Enter your blog URL and select the RSS you want to use in Telescope"
+      prompt="Enter your blog URL and select the RSS you want to use in Telescope.
+      A blog URL (HTML) should be the root of your blog itself, and Telescope will find the link(s) of blog feed URL (XML) for you to select, such as:"
```
Really picky thing, but I think you should add `the` before the blog
```suggestion
      A blog URL (HTML) should be the root of your blog itself, and Telescope will find the link(s) of the blog feed URL (XML) for you to select, such as:"
```

cychu42 replied:
Thanks, but it won't apply after the new changes.

cychu42 commented:
OK. I updated it. It now looks like this:
![image](https://user-images.githubusercontent.com/83015577/204037980-7b83ab31-2f66-43a5-aca3-9cf4d2cc5062.png)

For the prompt examples, I used the same font size as the `formControlLabel` class (the RSS URL after validation), which is smaller than the prompt.

manekenpix reviewed (approved):
LGTM :+1:
